### PR TITLE
Fix environment building on Cygwin.

### PR DESCRIPTION
**Pull request by dekimir**

### CODE REVIEW

**Comment by dekimir** [comment]
@AWoloszyn @antiagainst @dneto0 @Qining PTAL

File: utils/build.py [hunk 1]
```diff
@@ -104,12 +99,14 @@ def build(args):
                      '-DCMAKE_MAKE_PROGRAM=%s' % ninja]
 
     env = None
-    if (OS == 'Windows' or OS.startswith('CYGWIN')):
-        p = subprocess.Popen(
+    if (OS == 'Windows'):
```
Comment (antiagainst) [inline utils/build.py]: redundant `()` in `if`-statement.

File: utils/build.py [hunk 2]
```diff
@@ -104,12 +99,14 @@ def build(args):
...
+        env = parse_output_into_env(
             '"%VS140COMNTOOLS%..\\..\\VC\\vcvarsall.bat" & set',
-            stdout=subprocess.PIPE, cwd=args.builddir, shell=True)
-        env = dict([tuple(line.split('='))
-                    for line in p.communicate()[0].splitlines()])
+            args.builddir)
+    if (OS.startswith('CYGWIN')):
```
Comment (antiagainst) [inline utils/build.py]: This should be `elif`?

File: utils/build.py [hunk 3]
```diff
@@ -49,18 +49,13 @@ def run(cmd, cwd, env, justprint):
...
-    indices list.
+def parse_output_into_env(command, cwd):
+    """Runs command in cwd, expecting it to produce name=value lines on
+    output.  Parses those lines into a dictionary object.
     """
-    quoted = [(quote(e) if i in indices else e)
-              for (i, e) in enumerate(command)]
-    return ' '.join(quoted)
+    p = subprocess.Popen(command, stdout=subprocess.PIPE, cwd=cwd, shell=True)
+    return dict([tuple(line.split('='))
```
Comment (dneto0) [inline utils/build.py]: split is wrong here.  
For example, 
  'FOO=BAR=BAZ'.split('=)
will make a list of 3 elements.  You should only split on the first equal sign.

**Comment by dneto0** [comment]
-1 for the split bug.

**Comment by dekimir** [comment]
Feedback addressed.

**Comment by antiagainst** [comment]
+1

**Comment by dekimir** [comment]
Replaced by #176.

**Pull request closed without merging**